applications/jonny5/src/j5_per_minute.erl

```diff
@@ -161,3 +168,18 @@ metadata(Request) ->
        ,{<<"callee_id_number">>, j5_request:callee_id_number(Request)}
        ,{<<"rate">>, RateObj}
       ]).
+
+-spec calculate_seconds(integer(), integer(), integer(), integer()) -> integer().
+calculate_seconds(_, _, _, 0) ->  0;
+calculate_seconds(NoChargeTime, 0, Minimum, BillingSecs) ->
+    calculate_seconds(NoChargeTime, 60, Minimum, BillingSecs);
+calculate_seconds(NoChargeTime, Increment, Minimum, BillingSecs) ->
+    case BillingSecs =< 0 of
+        'true' -> 0;
+        'false' when BillingSecs =< NoChargeTime ->
+            0;
+        'false' when BillingSecs =< Minimum ->
+            Minimum;
+        'false' ->
+            kz_util:ceiling( BillingSecs / Increment ) * Increment
+    end.
```

Comment:
Hi @lazedo 

AFAIK NoChargeTime is a kind of a protection for customer against wrong (or maybe accidentally) dialed number or any other problem.
It's a minimal number of seconds needed to understand that something is wrong and hangup because of that. So, if your dialed number was wrong and you was quick enough to hangup you would pay nothing, if it was a normal call you pay in full.
So, from this point of view, calculations are correct.
In your example 23 seconds call should be belled as 23 seconds call, but 9 seconds call should be 0.

Regards,